src/packet_analysis/protocol/tcp/TCP.cc

```diff
@@ -96,6 +96,10 @@ void TCPAnalyzer::DeliverPacket(Connection* c, double t, bool is_orig, int remai
 	{
 	const u_char* data = pkt->ip_hdr->Payload();
 	int len = pkt->ip_hdr->PayloadLen();
+	// If the header length is zero, tcp checksum offloading is probably enabled
+	// In this case, let's fix up the length.
+	if ( pkt->ip_hdr->TotalLen() == 0 )
```

Comment:
won't make a functional difference but I'm wondering if this should check against `len` instead because that's the method now explicitly written to account for the offloading case.

Reply:
Hm, I am not sure. TotalLen is more reliable here - since there are other, legitimate cases in which the PayloadLength can be 0 (like a TCP packet with... well, no payload).